### PR TITLE
feat: IGL save resume

### PR DESCRIPTION
**Pull request by cheng-tan**

### CODE REVIEW

File: vowpalwabbit/core/include/vw/core/reductions/interaction_ground.h [hunk 1]
```diff
@@ -20,5 +20,29 @@ namespace reductions
...
+
+  std::unique_ptr<VW::workspace> ik_all;
+  ftrl* ik_ftrl;  // automatically save resume
+  std::unique_ptr<ftrl> pi_ftrl;
+};
 }  // namespace reductions
+
+namespace model_utils
+{
+size_t write_model_field(io_buf&, VW::reductions::ftrl&, const std::string&, bool);
```
Comment (bassmang) [inline vowpalwabbit/core/include/vw/core/reductions/interaction_ground.h:43]: can this be added to ftrl.h/cc instead?

File: vowpalwabbit/core/src/reductions/interaction_ground.cc [hunk 2]
```diff
@@ -23,6 +24,46 @@ using namespace VW::LEARNER;
 using namespace VW::config;
 using namespace VW::reductions;
 
+namespace VW
+{
+namespace reductions
+{
+namespace model_utils
```
Comment (bassmang) [inline vowpalwabbit/core/src/reductions/interaction_ground.cc:31]: May be useful to add a couple simple run_test's using -i/-f and --readable_model to make it clear what igl is saving

File: test/run_tests.py [hunk 3]
```diff
@@ -770,7 +770,7 @@ def convert_tests_for_flatbuffers(
             )
             continue
         # todo: 300 understand why is it failing
-        # test 189, 312, 316, 318, 351, 438, 394 and 319 depend on dsjson parser behaviour
+        # test 189, 312, 316, 318, 319, 351, 367, 368, 394, 438, 456, 457 depend on dsjson parser behaviour
```
Comment (bassmang) [inline test/run_tests.py:773]: 367/368 dont seem to be using dsjson?

Comment (cheng-tan) [reply test/run_tests.py:773]: https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/core.vwtest.json#L4736 I updated those two tests with the IGL reduction PR, but didn't update this comment.

**Review by bassmang — APPROVED** [review]

File: vowpalwabbit/core/src/reductions/ftrl.cc [hunk 4]
```diff
@@ -32,6 +33,31 @@ using namespace VW::reductions;
 #define W_WE 4  // Wealth
 #define W_MG 5  // maximum gradient
 
+namespace VW
+{
+namespace reductions
+{
+namespace model_utils
+{
+size_t write_model_field(io_buf& io, ftrl& ftrl_data, const std::string& upstream_name, bool text)
```
Comment (jackgerrits) [inline vowpalwabbit/core/src/reductions/ftrl.cc:42]: ftrl already has save resume, how is this different to what ftrl already does today?

**Pull request merged**